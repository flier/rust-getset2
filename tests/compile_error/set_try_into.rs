use getset2::Setter;

#[derive(Setter)]
pub struct Struct {
    #[set(try_into)]
    field: f32,
}

fn main() {
    let mut s = Struct { field: 3.14 };

    s.set_field(123).unwrap();
}
