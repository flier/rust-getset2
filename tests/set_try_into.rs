use getset2::{Getter, Setter};

#[derive(Default, Getter, Setter)]
pub struct Foo {
    #[get(copy)]
    #[set(try_into)]
    try_into_field: i32,
}

#[test]
fn set_try_into() {
    let mut foo = Foo::default();

    assert_eq!(foo.set_try_into_field(123).unwrap().try_into_field(), 123);
}
