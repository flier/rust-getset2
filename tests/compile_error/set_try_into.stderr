error[E0277]: the trait bound `f32: From<i32>` is not satisfied
  --> tests/compile_error/set_try_into.rs:12:7
   |
12 |     s.set_field(123).unwrap();
   |       ^^^^^^^^^ the trait `From<i32>` is not implemented for `f32`
   |
   = help: the following other types implement trait `From<T>`:
             <f32 as From<bool>>
             <f32 as From<i8>>
             <f32 as From<i16>>
             <f32 as From<u8>>
             <f32 as From<u16>>
   = note: required for `i32` to implement `Into<f32>`
   = note: required for `f32` to implement `TryFrom<i32>`
